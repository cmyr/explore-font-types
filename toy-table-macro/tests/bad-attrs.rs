toy_table_macro::tables! {
    TooMuchCount<'a> {
        item_count: Uint16,
        #[count(item_count)]
        #[count_with(double, item_count)]
        items: [Uint16],
    }
}

fn double(val: raw_types::Uint16) -> usize {
    val.get() as usize * 2
}

toy_table_macro::tables! {
    BadCountFn<'a> {
        item_count: Uint16,
        #[count_with(double, Self::item_count)]
        items: [Uint16],
    }
}

toy_table_macro::tables! {
    MissingCount<'a> {
        item_count: Uint24,
        items: [Uint24],
    }
}

toy_table_macro::tables! {
    CountOnScalar<'a> {
        #[count(item_count)]
        item_count: Uint24,
        #[count(item_count)]
        items: [Uint24],
    }
}

toy_table_macro::tables! {
    HiddenOnArray<'a> {
        item_count: Uint24,
        #[hidden]
        #[count(item_count)]
        items: [Uint24],
    }
}


fn main() {}
