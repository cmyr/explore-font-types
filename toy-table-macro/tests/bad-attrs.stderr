error: proc macro panicked
 --> tests/bad-attrs.rs:1:1
  |
1 | / toy_table_macro::tables! {
2 | |     TooMuchCount<'a> {
3 | |         item_count: Uint16,
4 | |         #[count(item_count)]
... |
7 | |     }
8 | | }
  | |_^
  |
  = help: message: I ONLY COUNT ONCE

error: expected ident
  --> tests/bad-attrs.rs:17:30
   |
17 |         #[count_with(double, Self::item_count)]
   |                              ^^^^

error: array types require 'count' or 'count_with' attribute
  --> tests/bad-attrs.rs:25:9
   |
25 |         items: [Uint24],
   |         ^^^^^

error: count/count_with attribute not valid on scalar fields
  --> tests/bad-attrs.rs:31:17
   |
31 |         #[count(item_count)]
   |                 ^^^^^^^^^^

error: 'hidden' is only valid on scalar fields
  --> tests/bad-attrs.rs:41:11
   |
41 |         #[hidden]
   |           ^^^^^^
